### PR TITLE
[MO] Clean up Model Optimizer options, help, and documentation

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** For several releases we have a bit dirty help and long-time ago dragged deprecated options. The following changes were done:
1. Remove the following Model Optimizer deprecated options that are no longer used for several releases: *--disable_gfusing*, *--generate_deprecated_IR_V7*, *--legacy_ir_generation*, *--keep_shape_ops, move_to_preprocess*
2. Deprecate through CLI the following options for which functionality triggered from POT or automatically: **--disable_weights_compression*, *--disable_nhwc_to_nchw*, *--disable_resnet_optimization*, *--finegrain_fusing*, *--disable_fusing*
3. Correct and extend description of each MO option to be printed during model conversion
4. Correct documentation about *--input_shape* since now we support dynamic dimensions and MO can determine input shapes on its own

**Tickets:** 79976, 80333

### CODE REVIEW

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 1]
```diff
@@ -371,23 +376,21 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
                                    'Available transformations: "LowLatency2", "MakeStateful"',
                               default="")
     common_group.add_argument('--disable_fusing',
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:378]: need to stay as deprecated since there are several tests for it

**Review by ilya-lavrenov — APPROVED** [review]

File: tools/mo/openvino/tools/mo/middle/fusings.py [hunk 2]
```diff
@@ -77,12 +77,11 @@ def find_and_replace_pattern(self, graph: Graph):
             for_graph_and_each_sub_graph_recursively(graph, fuse_linear_ops)
             for_graph_and_each_sub_graph_recursively(graph, lambda G: G.clean_up())
 
-        if not argv.disable_gfusing:
-            for_graph_and_each_sub_graph_recursively(graph, grouped_convolutions_fusing)
+        for_graph_and_each_sub_graph_recursively(graph, grouped_convolutions_fusing)
+        for_graph_and_each_sub_graph_recursively(graph, lambda G: G.clean_up())
+        if not argv.disable_fusing:
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/middle/fusings.py:82]: `disable_fusing` needs to stay as deprecated since we have mutliple Caffe tests with it.

**Review by mvafin — APPROVED** [review]

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 3]
```diff
@@ -371,23 +376,21 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
...
-                              action='store_true')
+                              help='[DEPRECATED] Turn off resnet optimization',
+                              action=DeprecatedStoreTrue, default=False)
     common_group.add_argument('--finegrain_fusing',
-                              help='Regex for layers/operations that won\'t be fused. ' +
-                                   'Example: --finegrain_fusing Convolution1,.*Scale.*')
+                              help='[DEPRECATED] Regex for layers/operations that won\'t be fused. ' +
+                                   'Example: --finegrain_fusing Convolution1,.*Scale.*',
+                              action=DeprecatedOptionCommon)
     common_group.add_argument('--disable_gfusing',
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:388]: why don't you delete it?

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 4]
```diff
@@ -371,23 +376,21 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
...
-                              help='Turn off fusing of grouped convolutions',
-                              action=DeprecatedStoreTrue)
+                              help=argparse.SUPPRESS,
+                              action=IgnoredAction)
     common_group.add_argument('--enable_concat_optimization',
-                              help='Turn on Concat optimization.',
-                              action='store_true')
-    common_group.add_argument('--move_to_preprocess',
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:388]: it was deprecated, can we just delete it?

**Review by jane-intel — CHANGES_REQUESTED** [review]

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 5]
```diff
@@ -371,23 +376,21 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
...
-                              help='Turn off resnet optimization',
-                              action='store_true')
+                              help='[DEPRECATED] Turn off resnet optimization',
+                              action=DeprecatedStoreTrue, default=False)
     common_group.add_argument('--finegrain_fusing',
-                              help='Regex for layers/operations that won\'t be fused. ' +
-                                   'Example: --finegrain_fusing Convolution1,.*Scale.*')
+                              help='[DEPRECATED] Regex for layers/operations that won\'t be fused. ' +
+                                   'Example: --finegrain_fusing Convolution1,.*Scale.*',
+                              action=DeprecatedOptionCommon)
```
[the remark this answers is not in the document]
Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py:387]: Sure, it should work. The action just prints about deprecation and the conversion goes on

File: tools/mo/openvino/tools/mo/utils/cli_parser.py (hunk 3 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py:388]: Agree, let us do this

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md [hunk 6]
```diff
@@ -218,7 +214,9 @@ There is no a universal recipe for determining the mean/scale values for a parti
...
 
 ## When to Specify Input Shapes <a name="when_to_specify_input_shapes"></a>
-There are situations when the input data shape for the model is not fixed, like for the fully-convolutional neural networks. In this case, for example, TensorFlow\* models contain `-1` values in the `shape` attribute of the `Placeholder` operation. Inference Engine does not support input layers with undefined size, so if the input shapes are not defined in the model, the Model Optimizer fails to convert the model. The solution is to provide the input shape(s) using the `--input` or `--input_shape` command line parameter for all input(s) of the model or provide the batch size using the `-b` command line parameter if the model contains just one input with undefined batch size only. In the latter case, the `Placeholder` shape for the TensorFlow\* model looks like this `[-1, 224, 224, 3]`. 
+There are situations when Model Optimizer is unable to deduce input shapes of the model, for example, in case of model pruning due to unsupported operations
```
[the remark this answers is not in the document]
Comment (rkazants) [reply docs/MO_DG/prepare_model/convert_model/Converting_Model.md:217]: Ok, I will correct

File: tools/mo/openvino/tools/mo/utils/cli_parser.py (hunk 4 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py:388]: According to the process let us deprecate this for 2022.1 release first. And after this we will remove it at all

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 7]
```diff
@@ -34,6 +34,13 @@ def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, self.dest, True)
 
 
+class DeprecatedOptionCommon(argparse.Action):
+    def __call__(self, parser, args, values, option_string=None):
```
[the remark this answers is not in the document]
Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py]: Done

File: tools/mo/openvino/tools/mo/utils/cli_parser.py (hunk 3 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py]: Done

File: tools/mo/openvino/tools/mo/utils/cli_parser.py (hunk 1 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py:378]: Not now please. We have a lot of them in testing repository.

**Review by jane-intel — APPROVED** [review]

**Review by tsavina — APPROVED** [review]

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md [hunk 8]
```diff
@@ -156,13 +153,12 @@ Framework-agnostic parameters:
                         original model is in FP32 and --data_type=FP16 is
                         specified, all model weights and biases are compressed
                         to FP16.
-  --disable_fusing      Turn off fusing of linear operations to Convolution
+  --disable_fusing      [DEPRECATED] Turn off fusing of linear operations to Convolution
```
Comment (andrew-zaytsev) [inline docs/MO_DG/prepare_model/convert_model/Converting_Model.md:156]:
```suggestion
  --disable_fusing      [DEPRECATED] Turn off fusing of linear operations to Convolution.
```

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md [hunk 9]
```diff
@@ -156,13 +153,12 @@ Framework-agnostic parameters:
...
   --disable_resnet_optimization
-                        Turn off resnet optimization
+                        [DEPRECATED] Turn off resnet optimization
```
Comment (andrew-zaytsev) [inline docs/MO_DG/prepare_model/convert_model/Converting_Model.md:158]:
```suggestion
                        [DEPRECATED] Turn off ResNet optimization.
```

**Review by andrew-zaytsev — CHANGES_REQUESTED** [review]

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md (hunk 9 above)
Comment (rkazants) [reply docs/MO_DG/prepare_model/convert_model/Converting_Model.md]: Done

**Review by andrew-zaytsev — APPROVED** [review]

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 10]
```diff
@@ -471,11 +464,12 @@ def get_common_cli_options(model_name):
...
     d['data_type'] = ['- Precision of IR', lambda x: 'FP32' if x == 'float' else 'FP16' if x == 'half' else x]
     d['disable_fusing'] = ['- Enable fusing', lambda x: not x]
-    d['disable_gfusing'] = ['- Enable grouped convolutions fusing', lambda x: not x]
-    d['move_to_preprocess'] = '- Move mean values to preprocess section'
+    d['transform'] = ['- User transformations', lambda x: x if x else 'Not specified']
     d['reverse_input_channels'] = '- Reverse input channels'
-    d['use_legacy_frontend'] = '- Use legacy API for model processing'
+    d['static_shape'] = '- Enable IR generation for fixed input shape'
     d['transformations_config'] = '- Use the transformations config file'
+    d['use_legacy_frontend'] = '- Force the usage of legacy API for model processing'
```
Comment (slyalin) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:471]: I afraid that term `legacy API` can be mixed with OV old API, but it is a different term. Can we provide more detailed explanation and separate it to a kind of advanced options? The same is about the next line.

**Review by slyalin — CHANGES_REQUESTED** [review]

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md [hunk 11]
```diff
@@ -258,10 +256,6 @@ Launch the Model Optimizer for the Caffe bvlc_alexnet model with specified input
 ```sh
 mo --input_model bvlc_alexnet.caffemodel --input "data[1 3 224 224]" --output pool5 -s 5 -b 8 --output_dir <OUTPUT_MODEL_DIR>
```
[the remark this answers is not in the document]
Comment (jane-intel) [reply docs/MO_DG/prepare_model/convert_model/Converting_Model.md:257]: this won't work. We cut with shape setting and than use -b?!?!

**Review by tgubanova-lohika — CHANGES_REQUESTED** [review]
Original comment was removed because it violates [A Code of Conduct for Open Source Communities](https://www.contributor-covenant.org/version/2/1/code_of_conduct/)

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (rkazants) [reply docs/MO_DG/prepare_model/convert_model/Converting_Model.md]: @slyalin, agree with you. I expect it to be a part of separate activity/PR. Now I am cleaning-up the current approach and the target of this PR is different that is to remove or deprecate redundant options, etc.

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md [hunk 12]
```diff
@@ -218,7 +214,9 @@ There is no a universal recipe for determining the mean/scale values for a parti
...
 
 ## When to Specify Input Shapes <a name="when_to_specify_input_shapes"></a>
-There are situations when the input data shape for the model is not fixed, like for the fully-convolutional neural networks. In this case, for example, TensorFlow\* models contain `-1` values in the `shape` attribute of the `Placeholder` operation. Inference Engine does not support input layers with undefined size, so if the input shapes are not defined in the model, the Model Optimizer fails to convert the model. The solution is to provide the input shape(s) using the `--input` or `--input_shape` command line parameter for all input(s) of the model or provide the batch size using the `-b` command line parameter if the model contains just one input with undefined batch size only. In the latter case, the `Placeholder` shape for the TensorFlow\* model looks like this `[-1, 224, 224, 3]`. 
+There are situations when Model Optimizer is unable to deduce input shapes of the model, for example, in case of model cutting due to unsupported operations
+or the model includes inputs of a dynamic rank. The solution is to provide input shapes of a static rank that can contain undefined dimensions `-1`
```
[the remark this answers is not in the document]
Comment (rkazants) [reply docs/MO_DG/prepare_model/convert_model/Converting_Model.md:218]: Agreed.

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (rkazants) [reply docs/MO_DG/prepare_model/convert_model/Converting_Model.md]: Corrected.

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 13]
```diff
@@ -417,19 +417,14 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
                                    'It will be DEPRECATED in future releases. '
                                    'Use --input option to specify a value for freezing.',
                               default=None)
-    common_group.add_argument('--generate_deprecated_IR_V7',
-                              help=argparse.SUPPRESS, action=IgnoredAction, default=False)
     common_group.add_argument('--static_shape',
                               help='Enables IR generation for fixed input shape (folding `ShapeOf` operations and '
                                    'shape-calculating sub-graphs to `Constant`). Changing model input shape using '
                                    'the Inference Engine API in runtime may fail for such an IR.',
```
[the remark this answers is not in the document]
Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py]: Done.

File: tools/mo/openvino/tools/mo/utils/cli_parser.py (hunk 10 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py]: Done.

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (rkazants) [reply docs/MO_DG/prepare_model/convert_model/Converting_Model.md]: Done.

**Review by slyalin — APPROVED** [review]

**Pull request merged**